Framework/Algorithms/src/ConvertSpectrumAxis2.cpp

```diff
@@ -232,39 +239,40 @@ MatrixWorkspace_sptr ConvertSpectrumAxis2::createOutputWorkspace(
   return outputWorkspace;
 }
 
-double
-ConvertSpectrumAxis2::getEfixed(const Mantid::Geometry::IDetector &detector,
-                                MatrixWorkspace_const_sptr inputWS,
-                                int emode) const {
+double ConvertSpectrumAxis2::getEfixed(
+    const size_t detectorIndex, const API::DetectorInfo &detectorInfo,
+    const Mantid::API::MatrixWorkspace &inputWS, const int emode) const {
   double efixed(0);
   double efixedProp = getProperty("Efixed");
+  Mantid::detid_t detectorID = detectorInfo.detectorIDs()[detectorIndex];
   if (efixedProp != EMPTY_DBL()) {
     efixed = efixedProp;
-    g_log.debug() << "Detector: " << detector.getID() << " Efixed: " << efixed
+    g_log.debug() << "Detector: " << detectorID << " Efixed: " << efixed
                   << "\n";
   } else {
     if (emode == 1) {
-      if (inputWS->run().hasProperty("Ei")) {
-        efixed = inputWS->run().getLogAsSingleValue("Ei");
+      if (inputWS.run().hasProperty("Ei")) {
+        efixed = inputWS.run().getLogAsSingleValue("Ei");
       } else {
         throw std::invalid_argument("Could not retrieve Efixed from the "
                                     "workspace. Please provide a value.");
       }
     } else if (emode == 2) {
-      std::vector<double> efixedVec = detector.getNumberParameter("Efixed");
+
+      const auto &detectorSingle = detectorInfo.detector(detectorIndex);
+
+      std::vector<double> efixedVec =
+          detectorSingle.getNumberParameter("Efixed");
       if (efixedVec.empty()) {
```

Comment:
You're right about the `DetectorGroup`, but I think that `Component::getNumberParameter` is also capable of returning something empty for the plain `Detector`.